### PR TITLE
ARM template instructions

### PR DESCRIPTION
**Pull request by Muksvso**
A document explaining 
1) how to create the Enterprise App registration with CLI and Manifest file changes. 
2) Using CLI to run the provided ARM template to get the experimentation setup deployed.

Additionally checks in the ARM template  , Parameters file for the template and, manifest file for the Entra ID App registration

### CODE REVIEW

File: ARMTemplateInstructions.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ms.author: mubatra
+ms.service: azure-app-configuration
+ms.topic: how-to 
+ms.date: 22/03/2024
+ms.custom: template-how-to
+---
+
+# Use ARM template for Experimentation resources
+
+This article will help you set up the Experimentation pipeline depicted in Figure (). You will be using Azure ClI and ARM template for majority of the setup.
```
Comment (maud-lv) [inline ARMTemplateInstructions.md:14]: what is Figure ()?

Comment (maud-lv) [reply ARMTemplateInstructions.md:14]: CLI

File: ARMTemplateInstructions.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		az ad sp create --id <objectid>
+	```
+- Check for the created application registration in **Entra Id -> App Registrations** on Azure portal. Go to **Manifest** blade, update the 'isEnabled' on line 35 to false, save the manifest. Copy the following values from the manifest,  `id` at line2, `appId` at line 7 and `name` at line 26, keep these values for use in next steps.
+
+	![Manifest](./Images/ManifestMenu.png)
+
+- Replace the entire manifest with the content copied from manifest.json in the repo. Edit the manifest with the values copied in previous step. Following edits are to be made :
+    1. "id" on line 2
+    1. "appId" on line 7
+    1. Add the "appId" on line 43 resulting in `api://$appId`.
```
Comment (maud-lv) [inline ARMTemplateInstructions.md:45]: `api://<appId>`

File: ARMTemplateInstructions.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    1. "name" of the app registration on line 55.
+        Save the manifest. You should see a manifest updated message.
+
+- Copy the following snippet on line 81, in the `preAuthorizedApplications`, and click save. You should get a manifest updated message.
```
Comment (maud-lv) [inline ARMTemplateInstructions.md:49]: Replace "Copy" by "Add".

Comment (maud-lv) [inline ARMTemplateInstructions.md:49]: message says "Successfully updated application"

File: ARMTemplateInstructions.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+## ARM template to setup the Experimentation resources
+
+- Create a resource group for Experimentation, in the subscription enabled for Preview. Set the subscription as default subscription for the CLI.
+  
+	```azurecli-interactive
+		az group create --location <location> --name <resource_group_name>
+	```
+
+- Update the ExperimentationPreviewParams.json file for the ARM template in ExperimentationPreviewTemplate.json. In ExperimentationPreviewParams.json, edit the value for "SEWEntraApplicationId" with the `application id` saved in the App registration step. Update the "RGname" parameter to have the name of the resource group you created in previous step. Run the deployment command on cli.
```
Comment (maud-lv) [inline ARMTemplateInstructions.md:86]: Tell the user they should first save ExperimentationPreviewParams.json and ExperimentationPreviewTemplate.json on their machine.

File: ARMTemplateInstructions.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  
+	```azurecli-interactive
+	        az ad app create --display-name <appRegistrationName>
+	```
+- Output will show objectid in the oauth2Permissions. Copy the value for the object id and run the following command with the copied object id.
+  
+	```azurecli-interactive
+		az ad sp create --id <objectid>
+	```
+- Check for the created application registration in **Entra Id -> App Registrations** on Azure portal. Go to **Manifest** blade, update the 'isEnabled' on line 35 to false, save the manifest. Copy the following values from the manifest,  `id` at line2, `appId` at line 7 and `name` at line 26, keep these values for use in next steps.
```
Comment (maud-lv) [inline ARMTemplateInstructions.md:38]: Remove "Go to **Manifest** blade, update the 'isEnabled' on line 35 to false, save the manifest"

File: ARMTemplateInstructions.md (hunk 1 above)
Comment (Muksvso) [reply ARMTemplateInstructions.md:14]: These are already corrected.

File: ARMTemplateInstructions.md (hunk 2 above)
Comment (Muksvso) [reply ARMTemplateInstructions.md:45]: not sure what is meant here?

File: ARMTemplateInstructions.md (hunk 4 above)
Comment (Muksvso) [reply ARMTemplateInstructions.md:86]: They are told to download or clone the repo in the start.

File: ARMTemplateInstructions.md (hunk 5 above)
Comment (Muksvso) [reply ARMTemplateInstructions.md:38]: will update to capture your experience as well, above is also correct experience so will keep both.

File: ARMTemplateInstructions.md (hunk 2 above)
Comment (Muksvso) [reply ARMTemplateInstructions.md:45]: got it.

**Pull request merged**